main.go

```diff
@@ -174,7 +175,14 @@ func generateEngine() *gin.Engine {
 		panic(err)
 	}
 
-	r := gin.Default()
+	// Initalize Gin
+	gin.SetMode(gin.ReleaseMode)
+	// Instantiate a new engine
+	r := gin.New()
+	// Use our custom logger
+	r.Use(middleware.DefaultStructuredLogger())
+	// Still use the built-in recovery middleware that is called with default
+	r.Use(gin.Recovery())
 
 	r.Use(
```

Comment:
can we merge all middleware's into this call so the stack order is clear? atm i'm not sure what runs first/last